src/Symfony/Component/Messenger/Middleware/ValidationMiddleware.php

```diff
@@ -28,9 +31,16 @@ public function __construct(ValidatorInterface $validator)
 
     public function handle($message, callable $next)
     {
-        $violations = $this->validator->validate($message);
+        $envelope = Envelope::wrap($message);
+        $groups = null;
+        /** @var ValidationConfiguration|null $validationConfig */
+        if ($validationConfig = $envelope->get(ValidationConfiguration::getName())) {
+            $groups = $validationConfig->getGroups();
+        }
+
+        $violations = $this->validator->validate($envelope->open(), null, $groups);
         if (\count($violations)) {
-            throw new ValidationFailedException($message, $violations);
+            throw new ValidationFailedException($envelope->open(), $violations);
```

Comment:
```diff
+        $originMessage = $envelope->open();
+        $violations = $this->validator->validate($originMessage, null, $groups);
         if (\count($violations)) {
-            throw new ValidationFailedException($message, $violations);
+            throw new ValidationFailedException($originMessage, $violations);
```
avoiding two calls?

Reply:
I changed that multiple times actually 😅 
I think I prefer keeping it like this to avoid any confusion between `$originMessage` and `$message`.

Reply:
call it `$subject`?

Reply:
Thanks, that's nice :)